storage/index/iterator.go

```diff
@@ -100,7 +93,13 @@ func (i *idsIter) Err() error {
 	return i.err
 }
 
-func (i *idsIter) release() {
+func (i *idsIter) Finalize() {
+	i.releaseCurrent()
+	*i = idsIterZeroed
+	// TODO(prateek): implement idsIter pooling
```

Comment:
As discussed, if we can pool the posting list returned that's probably most important first.